### PR TITLE
[Hydra] ConstraintViolationList documentation fix

### PR DESCRIPTION
**Pull request by fmata**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #930
| License       | MIT

### CODE REVIEW

File: src/Hydra/Serializer/DocumentationNormalizer.php [hunk 1]
```diff
@@ -396,13 +396,13 @@ private function getClasses(array $entrypointProperties, array $classes): array
                 [
                     '@type' => 'hydra:SupportedProperty',
                     'hydra:property' => [
-                        '@id' => '#ConstraintViolationList/violation',
-                        '@type' => 'rdf:Property',
-                        'rdfs:label' => 'violation',
+                        '@id' => '#ConstraintViolationList/violations',
+                        '@type' => 'hydra:Link',
```
Comment (dunglas) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: Are you sure about this change? We include the document so it's not a link (a link should refers to an IRI, not to an embedded document IIRC).

**Comment by dunglas** [comment]
Can you target the 2.0 branch please as it's a bug fix?

File: src/Hydra/Serializer/DocumentationNormalizer.php (hunk 1 above)
Comment (fmata) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: @dunglas I looked for more informations but I found nothing. May be I am wrong but I don't know how to process a property as relation if `@type` is not `hydra:Link`

Comment (dunglas) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: You can detect that it's an embedded relation because children documents contain an `@id` key.

Comment (dunglas) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: @lanthaler sorry to annoy you again but can you confirm that `Link` should not be used when the relation is embedded as a sub document?

Comment (fmata) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: @dunglas you are right, I can check if the targeted range is a supported class or not to guess if the property is a relation, and no more depend on `@type` of the property. It works fine. Tell me when you want I update my PR to keep `rdf:Property`.

Comment (dunglas) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: Yes please.

**Comment by fmata** [comment]
@dunglas done. Failed tests are unrelated.

**Comment by dunglas** [comment]
Tests fixed in #937. Can you rebase please?

**Comment by fmata** [comment]
@dunglas tests green :+1:

**Comment by dunglas** [comment]
Thanks @fmata

**Comment by lanthaler** [comment]
That’s fine. The embedded node can be used to warm up a cache for instance. It's a non-authoritative optimization though that the client might choose to ignore.

File: src/Hydra/Serializer/DocumentationNormalizer.php (hunk 1 above)
Comment (lanthaler) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: Using `Link` when the resource is embedded is fine. The embedded node can be used to warm up a cache for instance. As it is a non-authoritative optimization the client might choose to ignore though.

**Pull request merged**